ouroboros-network/src/Ouroboros/Network/PeerSelection/Governor/Types.hs

```diff
@@ -647,17 +724,28 @@ newtype Completion m peeraddr peerconn =
 data TracePeerSelection peeraddr =
        TraceLocalRootPeersChanged (LocalRootPeers peeraddr)
                                   (LocalRootPeers peeraddr)
+     -- | Peer selection targets changed: old targets, new targets.
      | TraceTargetsChanged     PeerSelectionTargets PeerSelectionTargets
+
```

Comment:
We should add a `TraceInsaneTargets`.

Reply:
:thinking: we could move the check outside of the outbound governor, and have a method which submits only if they are sane. But this is outside of this PR.